server/src/internalClusterTest/java/org/elasticsearch/indices/DataStreamIT.java

```diff
@@ -433,6 +444,87 @@ public void testCannotDeleteComposableTemplateUsedByDataStream() throws Exceptio
         assertTrue(maybeE.isPresent());
     }
 
+    public void testChangeTimestampFieldInComposableTemplatePriorToRollOver() throws Exception {
+        createIndexTemplate("id1", "logs-foo*", "@timestamp");
+
+        // Index doc that triggers creation of a data stream
+        IndexRequest indexRequest = new IndexRequest("logs-foobar").source("{}", XContentType.JSON).opType("create");
+        IndexResponse indexResponse = client().index(indexRequest).actionGet();
+        assertThat(indexResponse.getIndex(), equalTo(".ds-logs-foobar-000001"));
```

Comment:
Where possible, could we use the `DataStream::getDefaultBackingIndexName` method so that any changes there are automatically picked up at least in Java classes?